state/cockroachdb/cockroachdb.go

```diff
@@ -15,111 +15,111 @@ package cockroachdb
 
 import (
 	"context"
-	"reflect"
+	"fmt"
 
-	"github.com/dapr/components-contrib/metadata"
+	"github.com/jackc/pgx/v5"
+
+	"github.com/dapr/components-contrib/internal/component/postgresql"
 	"github.com/dapr/components-contrib/state"
 	"github.com/dapr/kit/logger"
 )
 
-// CockroachDB state store.
-type CockroachDB struct {
-	features []state.Feature
-	logger   logger.Logger
-	dbaccess dbAccess
-}
-
 // New creates a new instance of CockroachDB state store.
 func New(logger logger.Logger) state.Store {
-	dba := newCockroachDBAccess(logger)
-
-	return internalNew(logger, dba)
-}
-
-// internalNew creates a new instance of a CockroachDB state store.
-// This unexported constructor allows injecting a dbAccess instance for unit testing.
-func internalNew(logger logger.Logger, dba dbAccess) *CockroachDB {
-	return &CockroachDB{
-		features: []state.Feature{state.FeatureETag, state.FeatureTransactional, state.FeatureQueryAPI},
-		logger:   logger,
-		dbaccess: dba,
+	return postgresql.NewPostgreSQLStateStore(logger, postgresql.Options{
+		ETagColumn:    "etag",
+		EnsureTableFn: ensureTables,
+		SetQueryFn: func(req *state.SetRequest, opts postgresql.SetQueryOptions) string {
+			// Sprintf is required for table name because sql.DB does not substitute
+			// parameters for table names.
+			// Other parameters use sql.DB parameter substitution.
+			if req.ETag == nil || *req.ETag == "" {
+				return fmt.Sprintf(`
```

Comment:
As mentioned, consider using string concatenation rather than fmt.Sprintf for concatenating strings **in hot paths**. The perf difference is really significant.